### PR TITLE
[w32handle] Unify WaitHandle.Wait{One,Any,All} icalls

### PR DESCRIPTION
**Pull request by luhenry**

### CODE REVIEW

File: mono/metadata/threads.c [hunk 1]
```diff
@@ -1868,28 +1868,48 @@ map_native_wait_result_to_managed (MonoW32HandleWaitRet val, gsize numobjects)
...
-	start = (ms == -1) ? 0 : mono_100ns_ticks ();
-	do {
+	error_init (&error);
+
+	thread = mono_thread_internal_current ();
+
+	mono_thread_set_state (thread, ThreadState_WaitSleepJoin);
+
+	if (timeout == -1)
+	{
```
Comment (vargaz) [inline mono/metadata/threads.c]: This should use K&R indentation.

File: mono/metadata/threads.c [hunk 2]
```diff
@@ -1868,28 +1868,48 @@ map_native_wait_result_to_managed (MonoW32HandleWaitRet val, gsize numobjects)
 	}
 }
 
-static MonoW32HandleWaitRet
-mono_wait_uninterrupted (MonoInternalThread *thread, guint32 numhandles, gpointer *handles, gboolean waitall, gint32 ms, MonoError *error)
+gint32
+ves_icall_System_Threading_WaitHandle_Wait_internal (gpointer *handles, gint32 numhandles, MonoBoolean waitall, gint32 timeout)
```
Comment (kumpera) [inline mono/metadata/threads.c]: Shouldn't we take the opportunity to convert this icall to new style that has a `MonoError *error` argument ?

File: mono/metadata/w32handle.c [hunk 3]
```diff
@@ -282,36 +280,6 @@ mono_w32handle_init (void)
 	initialized = TRUE;
 }
 
-void
-mono_w32handle_cleanup (void)
```
Comment (kumpera) [inline mono/metadata/w32handle.c]: This change doesn't belong to a PR named "Unify WaitHandle.Wait{One,Any,All} icalls"

File: mono/metadata/w32socket-unix.c [hunk 4]
```diff
@@ -130,25 +126,6 @@ mono_w32socket_initialize (void)
 	mono_w32handle_register_ops (MONO_W32HANDLE_SOCKET, &ops);
 }
 
-static gboolean
-cleanup_close (gpointer handle, gpointer data, gpointer user_data)
```
Comment (kumpera) [inline mono/metadata/w32socket-unix.c]: I understand the reasoning for removing handles cleanup, I don't know know about this one.

Besides that. The commit messages for both removals are crap, they *must* state the reasoning behind the removal.

**Comment by luhenry** [comment]
@kumpera I updated according to your review

**Comment by Therzok** [comment]
Thanks for this PR, it hit me with an idea for gtk#!

**Pull request merged**